test/libyul/yulOptimizerTests/loopInvariantCodeMotion/no_move_state_loop.yul

```diff
@@ -0,0 +1,29 @@
+{
+  function f() -> x { x := g() }
+  function g() -> x { for {} 1 {} {} }
+
+  let b := 1
+  for { let a := 1 } iszero(eq(a, 10)) { a := add(a, 1) } {
+    let t := extcodesize(f())
+    let q := g()
+  }
+}
+// ----
+// step: loopInvariantCodeMotion
+//
+// {
+//     function f() -> x
+//     { x := g() }
+//     function g() -> x_1
+//     {
+//         for { } 1 { }
+//         { }
+//     }
+//     let b := 1
+//     let a := 1
+//     for { } iszero(eq(a, 10)) { a := add(a, 1) }
+//     {
+//         let t := extcodesize(f())
```

Comment:
Why isn't it moved?

Reply:
`f` causes an infinite loop and the for loop condition might be false

Reply:
Yea, somehow missed the condition in my head